dependencies.yaml

```diff
@@ -509,7 +509,7 @@ dependencies:
           - *scikit_learn
           - statsmodels
           - umap-learn==0.5.3
-          - pynndescent==0.5.8
+          - pynndescent
```

Comment:
@dantegd , is it alright if we relax this pin?

Reply:
Given passing tests, it should be fine to relax now